core/src/main/java/org/apache/iceberg/rest/requests/UpdateTableRequest.java

```diff
@@ -35,21 +35,22 @@
 public class UpdateTableRequest implements RESTRequest {
 
   private TableIdentifier identifier;
-  private List<UpdateRequirement> requirements;
+  private List<org.apache.iceberg.UpdateRequirement> requirements;
   private List<MetadataUpdate> updates;
 
   public UpdateTableRequest() {
     // needed for Jackson deserialization
   }
 
-  public UpdateTableRequest(List<UpdateRequirement> requirements, List<MetadataUpdate> updates) {
+  public UpdateTableRequest(
+      List<org.apache.iceberg.UpdateRequirement> requirements, List<MetadataUpdate> updates) {
     this.requirements = requirements;
     this.updates = updates;
   }
 
-  UpdateTableRequest(
+  public UpdateTableRequest(
```

Comment:
this is required to be public so that it can be used in `RESTSessionCatalog`. 
It might be also worth adding a Builder to `UpdateTableRequest` so that we don't have to make it public here.
In this case we have two options:
* write a manual builder
* generate a builder from the constructor definition (I've opened https://github.com/apache/iceberg/pull/7838 to indicate how this would look like)

Reply:
Maybe we should have a `create` method like `TableCommit`? Or one that accepts a `TableCommit` and returns the correct request? That would be simpler than a builder.

Reply:
sure that would also work. I'll make the respective changes in the next PR

Reply:
added in https://github.com/apache/iceberg/pull/7867